### ISSUE TITLE
twisted/words/test/test_service.py uses `deferredGenerator`

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6315]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)| @tomprince reported|
|-|-|
|Trac ID|trac#6315|
|Type|enhancement|
|Created|2013-02-21 06:44:42Z|
|Branch|https://github.com/twisted/twisted/tree/de-defgen-twisted.words-6315|

[#6044](https://github.com/twisted/twisted/issues/6044) wants to deprecate `deferredGenerator`, but it is used extensively in t.w.test.test_service. It actually appears that (since the tests are using the synchronous `InMemoryWordsRealm`, that the tests could instead use `successResultOf` and `failureResultOf` instead.

<details><summary>Searchable metadata</summary>

```
trac-id__6315 6315
type__enhancement enhancement
reporter__tom_prince tom.prince
priority__normal normal
milestone__ 
branch__branches_de_defgen_twisted_words_6315 branches/de-defgen-twisted.words-6315
branch_author__tomprince tomprince
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1361429082000000 1361429082000000
changetime__1363629333000000 1363629333000000
version__None None
owner__tom_prince tom.prince
cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_2"></a>|@tomprince commented|
|-|-|

(In [37192]) Branching to de-defgen-twisted.words-6315.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)|@tomprince set owner to @ralphm|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_4"></a>|@wallrj set owner to @tomprince|
|-|-|

Here's a quick, inexpert, review of the changes:

 * I inspected all the changes and the updated tests all appear to be
   equivalent to the original tests.
 * It's nice that the new tests are so much more succinct.
 * All those new tests pass for me (Fedora 18)
 * I compared coverage of words.service in trunk against this branch and it
   appears to be exactly the same...which is reassuring.

   ```
--- /tmp/coverage_report_de-defgen.log  2013-03-09 23:45:47.034799667 +0000
+++ /tmp/coverage_report_trunk.log      2013-03-09 23:42:55.055202673 +0000
...
Name                                                                                           Stmts   Miss Branch BrMiss  Cover
--------------------------------------------------------------------------------------------------------------------------------
...
 twisted/words/ewords                                                                              15      1      0      0    93%
 twisted/words/iwords                                                                              40      0     44     22    74%
 twisted/words/protocols/__init__                                                                   0      0      0      0   100%
-twisted/words/protocols/irc                                                                     1280    758    364    315    35%
+twisted/words/protocols/irc                                                                     1279    758    364    315    35%
 twisted/words/service                                                                            601    101    118     33    81%
 twisted/words/test/__init__                                                                        0      0      0      0   100%
-twisted/words/test/test_service                                                                  523     12     34      4    97%
+twisted/words/test/test_service                                                                  682     12     34      4    98%
   ```


I noticed one issue:
 1. [One new build error](https://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/de-defgen-twisted.words-6315)
    1. There's a new line length warning
       ```
       ************* Module twisted.words.test.test_service
       C0301:106,0: Line too long (85/79)
       ```


You may prefer ralphm to have a look at the branch too, but it looks
good to me.

Fix that long line and merge as far as I'm concerned.

-RichardW.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_5"></a>|@tomprince set status to `closed`|
|-|-|

(In [37583]) Merge de-defgen-twisted.words-6315: Use inlineCallbacks in twisted.words.test.test_service.

Author: tom.prince
Reviewers: rwall
Fixes: [#6315](https://github.com/twisted/twisted/issues/6315)

We want to deprecate deferredGenerator, so remove the uses of
it here.
